ignite/handlers/lr_finder.py

```diff
@@ -119,12 +120,41 @@ def _run(
 
         self.logger.debug(f"Running LR finder for {num_iter} iterations")
         if start_lr is None:
-            start_lr = optimizer.param_groups[0]["lr"]
-        # Initialize the proper learning rate policy
-        if step_mode.lower() == "exp":
+            start_lr_list = [pg["lr"] for pg in optimizer.param_groups]
+        elif isinstance(start_lr, float):
             start_lr_list = [start_lr] * len(optimizer.param_groups)
-            self._lr_schedule = LRScheduler(_ExponentialLR(optimizer, start_lr_list, end_lr, num_iter))
+        elif isinstance(start_lr, list):
+            if len(start_lr) != len(optimizer.param_groups):
+                raise ValueError(
+                    f"Number of values of start_lr should be equal to optimizer values. start_lr values:{len(start_lr)} optimizer values: {len(optimizer.param_groups)}"
+                )
+            start_lr_list = start_lr
+        else:
+            raise TypeError(f"start_lr should a float or list of floats, but given {type(start_lr)}")
+        if isinstance(end_lr, float):
+            end_lr_list = [end_lr] * len(optimizer.param_groups)
+        elif isinstance(end_lr, list):
+            if len(end_lr) != len(optimizer.param_groups):
+                    raise ValueError(
+                        f"Number of values end_lr should be equal to optimizer values. end_lr values:{len(end_lr)} optimizer values: {len(optimizer.param_groups)}"
+                    )
+            end_lr_list = end_lr
+        else:
+            raise TypeError(f"end_lr should a float or list of floats, but given {type(end_lr)}")
+        
+        # Initialize the proper learning rate policy
+        if step_mode.lower() == "exp":       
+            self._lr_schedule = LRScheduler(_ExponentialLR(optimizer, start_lr_list, end_lr_list, num_iter))
         else:
+            if isinstance(start_lr, list) or isinstance(end_lr, list):
+                self._lr_schedule = ParamGroupScheduler([
+                PiecewiseLinear(
+                    optimizer,
+                    param_name="lr",
+                    milestones_values=[(0, start_lr_list[i]), (num_iter, end_lr_list[i])],
+                    param_group_index=i
+                ) for i in range(len(optimizer.param_groups))]
+            )
             self._lr_schedule = PiecewiseLinear(
```

Comment:
Errors are due to here in which `_lr_schedule` is incorrectly overriden. Please remove this line.